docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+   update can properly work using `copy-on-write <https://en.wikipedia.org/wiki/Copy-on-write>`_ principles.
+
+Below is an example where we update the existing dataset ``another_unique_dataset_identifier``
+with new data for ``table1`` and ``table2``:
+
+.. ipython:: python
+
+    another_df2 = pd.DataFrame(
+        {
+            "G": "bar",
+            "H": pd.Categorical(["prod", "dev", "prod", "dev"]),
+            "I": np.array([12] * 4, dtype="int32"),
+            "J": pd.Series(4, index=list(range(4)), dtype="float32"),
+            "K": pd.Timestamp("20190614"),
+            "L": 10.0,
+        }
+    )
+    another_df2
+
+    dm = update_dataset_from_dataframes(
+        {"data": {"table1": another_df, "table2": another_df2}},
+        store=store_factory,
+        dataset_uuid="another_unique_dataset_identifier",
+    )
+    dm.tables
+    sorted(dm.partitions.keys())
+
+
+Trying to update only a subset of tables throws a ``ValueError``:
+
+.. ipython::
+
+    @verbatim
+    In [45]: update_dataset_from_dataframes(
+       ....:        {
+       ....:           "data":
+       ....:           {
+       ....:              "table2": another_df2
+       ....:           }
+       ....:        },
+       ....:        store=store_factory,
+       ....:        dataset_uuid="another_unique_dataset_identifier"
+       ....:        )
+       ....:
+    ---------------------------------------------------------------------------
+    ValueError: Input partitions for update have different tables than dataset:
+    Input partition tables: {'table2'}
+    Tables of existing dataset: ['table1', 'table2']
+
+
+Adding data to an existing dataset is not the only functionality achievable within an update
+operation, such an operation can also be used to remove or overwrite data.
+To do so we use the ``delete_scope`` keyword argument as shown in the example below:
+
+.. ipython:: python
+
+    dm = update_dataset_from_dataframes(
+        None,
+        store=store_factory,
+        dataset_uuid="partitioned_dataset",
+        partition_on="E",
+        delete_scope=[{"E": "train"}],
+    )
+    sorted(dm.partitions.keys())
+
+
+As we can see, we specified using a dictionary that data where the column ``E`` has the
+value ``train`` should be removed. Looking at the partitions after the update, we see that
+the partition ``E=train`` has been removed.
+
+.. note:: We defined ``delete_scope`` over a value of ``E`` which is also the column that
+    we partitioned on: ``delete_scope`` only works *as expected* on indexed columns.
+
+    Furthermore it *should only* be used on partitioned columns due to their one-to-one mapping;
+    without the guarantee of one-to-one mappings, using ``delete_scope`` could have unwanted
+    effects like accidentally removing data with different values.
+
+
+.. admonition:: Using ``delete_scope`` on data not partitioned by columns
```

Comment:
Not entirely sure if this is a feature or a bug. I would consider it a bug if a delete scope is requested for a column for which we don't have index information

Reply:
Regardless of the label (bug vs feature) this is possible right now - ie. it doesn't throw an error; thus the admonition, to basically warn the user not use it for the purpose of deletion. Probably this too could benefit from being promoted to a warning.